assets/wire/actions.js

```diff
@@ -37,9 +38,9 @@ export function preview(item) {
 }
 
 export function previewAndCopy(item) {
-    return (dispatch) => {
+    return (dispatch, getState) => {
         dispatch(previewItem(item));
-        dispatch(copyPreviewContents(item));
+        dispatch(getState().context === 'agenda' ? copyPreviewContentsOfAgendaItem(item) : copyPreviewContents(item));
```

Comment:
I guess agenda should call different action and not something from wire

Reply:
Yes, In that case I need to refactor this bit to pass in different action, what do you think? https://github.com/superdesk/newsroom/blob/master/assets/wire/item-actions.js#L10

Reply:
yep imo the actions should be constructed in the main app component, wireApp or agendaApp